### PR TITLE
Reworking of WebClient

### PR DESCRIPTION
**Pull request by PierreSavatte**

### CODE REVIEW

File: kirby/api/ext.py [hunk 1]
```diff
@@ -177,31 +177,26 @@ def __exit__(self, exc_type, exc_val, exc_tb):
...
-    def post(self, endpoint, data, params=None):
-        if not params:
-            params = {}
-        result = self._session.post(
-            urljoin(self.web_endpoint_base, endpoint), data=data, params=params
-        )
+    def _request(self, method, endpoint, params):
+        url = urljoin(self.web_endpoint_base, endpoint)
+
+        if method == "GET":
```
Comment (ericgazoni) [inline kirby/api/ext.py]: https://tenor.com/FGRO.gif

**Comment by PierreSavatte** [comment]
Now support every verb that requests support

File: kirby/api/ext.py (hunk 1 above)
Comment (PierreSavatte) [reply kirby/api/ext.py]: Is it better now?

File: kirby/api/ext.py [hunk 2]
```diff
@@ -198,32 +198,24 @@ def __enter__(self):
...
-        )
-        if result.status_code == 200:
-            return result.json()
-        raise WebClientError(
-            f"POST error on {result.url}. "
-            f"Status code : {result.status_code}. "
-            f"Response : {result.text}"
-        )
+    def _request_decorator(self, method):
+        @tenacity.retry(**webserver_retry_args)
```
Comment (ericgazoni) [inline kirby/api/ext.py]: you do not **need** to use tenacity.retry as an `@decorator`, you could juste 

```python
def request(self, endpoint, method):
    return tenacity.retry(**webserver_retry_args)(method(...)
```

File: kirby/api/ext.py [hunk 3]
```diff
@@ -198,32 +198,24 @@ def __enter__(self):
...
+                return result.json()
+            raise WebClientError(
+                f"{method} error on {result.url}. "
+                f"Status code : {result.status_code}. "
+                f"Response : {result.text}"
+            )
+
+        return _request
+
+    def __getattr__(self, item):
```
Comment (ericgazoni) [inline kirby/api/ext.py]: Can you a test where you test the most common verbs (e.g. get, post, put, delete, head, options) please? Just to make sure they work (but obviously you can mock the actual calls)

File: kirby/api/ext.py (hunk 2 above)
Comment (PierreSavatte) [reply kirby/api/ext.py]: It is now written as you proposed

File: kirby/api/ext.py (hunk 3 above)
Comment (PierreSavatte) [reply kirby/api/ext.py]: Test written. Also, now testing errors handling with the same principle

**Pull request merged**